addons/website_crm/controllers/main.py

```diff
@@ -49,7 +49,21 @@ def website_form(self, model_name, **kwargs):
         return super(WebsiteForm, self).website_form(model_name, **kwargs)
 
     def insert_record(self, request, model, values, custom, meta=None):
+        Visitor = request.env['website.visitor']
         if model.model == 'crm.lead':
             if 'company_id' not in values:
                 values['company_id'] = request.website.company_id.id
-        return super(WebsiteForm, self).insert_record(request, model, values, custom, meta=meta)
+            lang = request.context.get('lang', False)
+            lang_id = request.env["res.lang"].sudo().search([('code', '=', lang)], limit=1).id
+            values['lang_id'] = lang_id
+        result = super(WebsiteForm, self).insert_record(request, model, values, custom, meta=meta)
```

Comment:
A bit generic question: shouldn't we take default_lang_id from website ? Or is it put into context ?

In short: is the lang correctly managed / propagated ?